plugins/dell-kestrel/fu-dell-kestrel-ec-hid.c

```diff
@@ -0,0 +1,141 @@
+/*
+ * Copyright 2024 Dell Technologies
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR MIT
+ */
+
+#include "config.h"
+
+#include <errno.h>
+#include <string.h>
+
+#include "fu-dell-kestrel-ec-hid.h"
+
+gboolean
+fu_dell_kestrel_ec_hid_write(FuDevice *device, GBytes *buf, GError **error)
+{
+	guint8 *data = (guint8 *)g_bytes_get_data(buf, NULL);
```

Comment:
this looks fishy; you're converting something that's immutable (`const guint8 *`) into (`guint8 *`).

It seems to me that you're reading the page `page = fu_chunk_array_index(pages, j, error);` and then padding it out to DELL_KESTREL_EC_HID_DATA_PAGE_SZ with 0xFF, then writing. I'd suggest something like:

    g_autoptr(GByteArray) page_aligned = g_byte_array_new();
    g_byte_array_append(page_aligned, fu_chunk_get_data(page), fu_chunk_get_data_sz(page));
    fu_byte_array_set_size(page_aligned, DELL_KESTREL_EC_HID_DATA_PAGE_SZ, 0xFF);

in `fu_dell_kestrel_ec_write_firmware_pages` -- and then pass the mutable GByteArray to fu_dell_kestrel_ec_hid_write() rather than the immutable GBytes.